sprints/dashboard/tests/test_utils.py

```diff
@@ -322,3 +327,70 @@ def test_column_number_to_excel(test_input, expected):
+        mock_get.return_value = Mock(text=f.read())
+        output = dict(get_cell_member_roles())
+        expected_output = {
+                            "Member Six": ["Recruitment manager"],
+                            "Member Seven": ["Sprint manager"],
+                            "Member Five": ["Sprint Planning Manager"],
+                            "Member Eight": ["Epic manager"],
+                            "Member Two": ["Sustainability manager"],
+                            "Member Four": ["OSPR Liaison", "Official forum moderator"],
+                            "Member One": ["DevOps Specialist"],
+                        }
+        TestCase().assertDictEqual(output, expected_output)
+
+@patch("requests.get")
+def test_get_cell_member_roles_corrupted(mock_get):
+    with open("sprints/dashboard/tests/data/handbook/dummy_cells_corrupted.html", 'r') as f:
+        mock_get.return_value = Mock(text=f.read())
+        with TestCase().assertRaises(NoRolesFoundException):
+            get_cell_member_roles(raise_exception=True)
+
+def test_get_rotations_roles_for_member():
+    # When the member is on FF duty
+    output = get_rotations_roles_for_member('John Doe', {'FF': ['Jake Doe', 'John Doe'], 'DD': ['Jane Doe', 'James Doe']})
+    assert len(output) == 1 and output[0] == 'FF-2'
+
+    # When member has no duty
+    output = get_rotations_roles_for_member('John Doe', {'FF': ['Jake Doe', 'James Doe'], 'DD': ['Jane Doe', 'James Doe']})
+    assert len(output) == 0
+
+    # When only partial name of the member is defined
+    output = get_rotations_roles_for_member('John Doe', {'FF': ['John Doe', 'Jake Doe'], 'DD': ['Jane Doe', 'James Doe']})
+    assert len(output) == 1 and output[0] == 'FF-1'
+
+def test_compile_participants_roles():
+    members_data_dummy = [
+        Mock(displayName='Jane Doe', emailAddress='janedoe@opencraft.com'),
+        Mock(displayName='Jack Doe', emailAddress='jackdoe@opencraft.com'),
+        Mock(displayName='John Doe', emailAddress='johndoe@opencraft.com'),
+        Mock(displayName='Jake Doe', emailAddress='jakedoe@opencraft.com'),
+    ]
+
+    rotations_data_dummy = {
+        'ff': ['John Doe', 'Jack'],
+        'dd': ['Jack Doe', 'Jake Doe'],
+    }
+
+    roles_data_dummy = {
+        'Jane Doe': [],
+        'Jack Doe': ['Sprint Planning Manager', 'DevOps Specialist'],
+        'John Doe': ['Sprint Manager'],
+        'Jake Doe': ['Recruitment Manager'],
+    }
+
+    output = compile_participants_roles(members_data_dummy, rotations_data_dummy, roles_data_dummy)
+
+    expected_output = {
+        'janedoe@opencraft.com': [],
+        'jackdoe@opencraft.com': ['Sprint Planning Manager', 'DevOps Specialist', 'ff-2', 'dd-1'],
+        'johndoe@opencraft.com': ['Sprint Manager', 'ff-1'],
+        'jakedoe@opencraft.com': ['Recruitment Manager', 'dd-2']
+    }
+
+    TestCase().assertDictEqual(output, expected_output)
```

Comment:
It might make sense to verify case with empty data too, to be sure that all corner cases will remain handled if we make any changes to these functions.

Reply:
@nizarmah, yep, that's what I meant :+1: